src/nspawn/nspawn.c

```diff
@@ -3453,6 +3453,38 @@ static int inner_child(
         if (r < 0)
                 return r;
 
+        if (arg_start_mode == START_BOOT) {
+                int check_os_release, is_os_tree;
+
+                check_os_release = getenv_bool("SYSTEMD_NSPAWN_CHECK_OS_RELEASE");
+                if (check_os_release < 0 && check_os_release != -ENXIO) {
+                        return log_error_errno(check_os_release, "Failed to parse $SYSTEMD_NSPAWN_CHECK_OS_RELEASE: %m");
+                }
+
+                is_os_tree = path_is_os_tree("/");
+                if (is_os_tree == 0 && check_os_release == 0)
+                        log_debug("Root tree is missing an os-release file, continuing anyway.");
+                else if (is_os_tree <= 0) {
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Root tree doesn't look correct (os-release file is missing). Refusing.");
+                }
```

Comment:
no {} around single line if blocks. as per coding style.

also, if you do have an error code, please log the error. i.e. if `is_os_tree` is < 0, that's an error code, hence please cover that in a separate branch and pass it as first arg to `log_error_errno()`, and suffix it with `: %m` to display the error code.